src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -109,10 +118,37 @@ private static string GetDotNetBasePath(string workingDirectory)
             var matched = DotNetBasePathRegex.Match(outputString);
             if (!matched.Success)
             {
-                return null;
+                yield break; 
             }
+            
+            var basePath = matched.Groups[1].Value.Trim();
 
-            return matched.Groups[1].Value.Trim();
+            yield return basePath; // We return the version in use at the front of the list in order to ensure FirstOrDefault always returns the version in use.
+
+            var lineSdkIndex = lines.FindIndex(line => line.Contains("SDKs installed"));
+
+            if (lineSdkIndex != -1)
+            {
+                lineSdkIndex++;
+
+                while (lineSdkIndex < lines.Count && !string.IsNullOrEmpty(lines[lineSdkIndex]))
+                {
+                    var sdkMatch = SdkRegex.Match(lines[lineSdkIndex]);
+
+                    if (!sdkMatch.Success)
+                        break;
+
+                    var version = sdkMatch.Groups[1].Value.Trim();
+                    var path = sdkMatch.Groups[2].Value.Trim();
+                    
+                    path = Path.Combine(path, version) + Path.DirectorySeparatorChar;
+
+                    if (!path.Equals(basePath))
+                        yield return path; 
```

Comment:
Part of me wants to order this list by version descending (`dotnet --info` returns results in ascending order). But we already extract the `global.json`-or-highest version as `basePath` first, and that ordering is most critical. And we don't order VS instances. So I think the current approach is good.

Reply:
I think it's still useful to sort the results even with the basePath first because of https://github.com/microsoft/MSBuildLocator/pull/106. If the default SDK is from after the runtime in use, it'll be disqualified, and we'd go with the second in the list. If you don't clean your machine regularly and still have a 1.0 SDK lying around, that's what we'd find and register with RegisterDefaults.

Reply:
I'm convinced.

Reply:
So should I remove the new iterator/yield code and go back to the code using List<string> with Reverse(), or do we have a better solution based on the current code?

Reply:
I think that's the best solution. I already wrote up the change in #106, so I wouldn't worry about remaking that change here.